nussl/core/augmentation.py

```diff
@@ -0,0 +1,377 @@
+    Returns:
+        shifted_signal: A copy of the original audio signal, with augmentations applied 
+    """
+    if not isinstance(shift, int):
+        raise ValueError("shift must be an integer.")
+
+    sample_rate = audio_signal.sample_rate
+    shifted_audio_data = []
+    audio_data = audio_signal.audio_data
+
+    for audio_row in audio_signal.get_channels():
+        shifted_audio_data.append(librosa.effects.pitch_shift(audio_row, sample_rate, shift))
+    shifted_audio_data = np.array(shifted_audio_data)
+    shifted_signal = audio_signal.make_copy_with_audio_data(shifted_audio_data)
+    shifted_signal.stft()
+
+    return shifted_signal
+
+def low_pass(audio_signal, highest_freq):
+    """
+    Applies low pass filter. 
+    Args: 
+        audio_signal: An AudioSignal object
+        highest_freq: Threshold for low pass. Should be positive scalar
+    Returns:
+        augmented_signal: A copy of the original audio signal, with augmentations applied 
+    """
+    if not np.issubdtype(type(highest_freq), np.number) or highest_freq <= 0:
+        raise ValueError("highest_freq should be positve scalar")
+    
+    l_stft = audio_signal.stft_data.copy()
+    freq_vector = audio_signal.freq_vector
+    idx = (np.abs(freq_vector - highest_freq)).argmin()
+    l_stft[idx:, :, :] = 0
+    augmented_signal = audio_signal.make_copy_with_stft_data(l_stft)
+    augmented_signal.istft()
+
+    return augmented_signal
+
+def high_pass(audio_signal, lowest_freq):
+    """
+    Applies high pass filter
+    Args: 
+        audio_signal: An AudioSignal object
+        highest_freq: Threshold for high pass. Should be positive scalar
+    Returns:
+        augmented_signal: A copy of the original audio signal, with augmentations applied 
+    """
+    if not np.issubdtype(type(lowest_freq), np.number) or lowest_freq <= 0:
+        raise ValueError("lowest_freq should be positve scalar")
+
+    h_stft = audio_signal.stft_data.copy()
+    freq_vector = audio_signal.freq_vector
+    idx = (np.abs(freq_vector - lowest_freq)).argmin()
+    h_stft[:idx, :, :] = 0
+    augmented_signal = audio_signal.make_copy_with_stft_data(h_stft)
+    augmented_signal.istft()
+
+    return augmented_signal
+
+def tremolo(audio_signal, mod_freq, mod_depth):
+    """
+    https://ffmpeg.org/ffmpeg-all.html#tremolo
+
+    Applies tremolo filter
+    Args: 
+        audio_signal: An AudioSignal object
+        mod_freq: Modulation frequency. Must be between 0 and 1.
+        mod_depth: Modulation depth. Must be between 0 and 1.
+    Returns:
+        augmented_signal: A copy of the original audio signal, with augmentations applied 
+    """
+    if not np.issubdtype(type(mod_freq), np.number) or mod_freq < 0:
+        raise ValueError("mod_freq should be positve scalar")
+    
+    if not np.issubdtype(type(mod_depth), np.number) or mod_depth < 0 or mod_depth > 1:
+        raise ValueError("mod_depth should be positve scalar between 0 and 1.")
+
+    filter_kwargs = {
```

Comment:
Why not do this directly? Why create a kwargs dict first then use it?

Reply:
So for every ffmpeg filter, I found myself writing the code found in lines 52-71 in `augmentation_utils.py` in every single function. The only difference in each filter is the kwargs for `ffmpeg.filter()`. So this became the primary input to `apply_ffmpeg_filter`. If I was not to use kwargs here, I would have to declare every single possible filter argument in `apply_ffmpeg_filter`.

Reply:
Instead of making a dictionary, you can pass them as keyword arguments to `apply_ffmpeg_filter`, instead of making a dictionary and then passing the dictionary.

Reply:
Just made that change in my code. I didn't actually know you could do that.